src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -890,6 +890,14 @@ VideoDecoder::DecodedOutput VideoDecoder::convertAVFrameToDecodedOutput(
   return output;
 }
 
+// Note [preAllocatedOutputTensor with swscale and filtergraph]:
+// Callers may pass a pre-allocated tensor, where the output frame tensor will
+// be stored. This parameter is honored in any case, but it only leads to a
+// speed-up when swscale is used. With swscale, we can tell ffmpeg to place the
+// decoded frame directly into `preAllocatedtensor.data_ptr()`. That's not
+// possible with filtegraph, leading to an extra copy.
```

Comment:
I don't think it's impossible with filtergraph -- we just don't know and I haven't investigated this further.

If ffmpeg allows us to pass in pre-allocated frames to filtergraph we can do this more efficiently for filtergraph too. I would add a TODO to investigate and fix this for filtergraph later.